gno.land/pkg/sdk/vm/convert.go

```diff
@@ -34,9 +41,7 @@ func convertArgToGno(arg string, argT gno.Type) (tv gno.TypedValue) {
 			tv.SetString(gno.StringValue(arg))
 			return
 		case gno.IntType:
-			if arg[0] == '+' {
-				panic("numbers cannot start with +")
-			}
+			assertCharNotPlus(arg[0])
```

Comment:
we can divide the case in two to avoid that amount of duplicated code:

```
switch bt {
	case gno.BoolType:
		if arg == "true" {
			tv.SetBool(true)
			return
		} else if arg == "false" {
			tv.SetBool(false)
			return
		} else {
			panic(fmt.Sprintf(
				"unexpected bool value %q",
				arg))
		}
	case gno.StringType:
		tv.SetString(gno.StringValue(arg))
		return
	default:
		return handleNumericTypes(...)
		
```

```
func handleNumericTypes(arg string, bt ...) gno.TypedValues {
	assertCharNotPlus(arg[0])
	select bt {
		case gno.IntType:
		 .... // you don't have to repeat the assert on every numeric case
	}
}

```

Still not sure why we need to check for a plus sign tho.

Reply:
That's true. I'm not sure why this explicit check must exist.  I added it to the float type checks to keep it consistent and moved the plus char check to a function to deduplicate the logic. I kind of like seeing all the types here in one switch case and would prefer not to move them to a switch in another function just to avoid deduplicating that function call.  I don't feel that strongly about this though